modApiServer/src/org/aion/api/server/pb/ApiAion0.java

```diff
@@ -1306,6 +1224,185 @@ public void onBlock(IBlockSummary cbs) {
+
+            try {
+                req = Message.req_getBlockDetailsByLatest.parseFrom(data);
+                Long count = req.getCount();
+                if (count > 1000) {
+                    count = 1000L;
+                }
+
+                Long latest = this.getBestBlock().getNumber();
+                List<Long> numbers = new ArrayList<>();
+                long i = Long.valueOf(latest);
+                while (i > Long.valueOf(latest)-Long.valueOf(count) && i >= 0) {
+                    numbers.add(i);
+                    i--;
+                }
+
+                List<Map.Entry<AionBlock, BigInteger>> blks =
+                        numbers.parallelStream()
+                                .filter(n -> this.getBlock(n)!= null)
+                                .map(n -> Map.entry(this.getBlock(n),
+                                        this.ac.getBlockchain().getTotalDifficultyByHash(
+                                                Hash256.wrap(this.getBlock(n).getHash()))))
+                                .collect(Collectors.toList());
+
+                if (blks == null) {
+                    return ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_fail_function_arguments_VALUE);
+                } else {
+                    List<Message.t_BlockDetail> bds = getRsp_getBlockDetails(blks);
+                    Message.rsp_getBlockDetailsByLatest rsp =  Message.rsp_getBlockDetailsByLatest.newBuilder()
+                            .addAllBlkDetails(bds)
+                            .build();
+
+                    byte[] retHeader = ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_success_VALUE);
+                    return ApiUtil.combineRetMsg(retHeader, rsp.toByteArray());
+                }
+            } catch (Exception e) {
+                LOG.error("ApiAionA0.process.getBlockDetailsByLatest exception: [{}]", e.getMessage());
+                return ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_fail_function_exception_VALUE);
+            }
+        }
+        case Message.Funcs.f_getBlocksByLatest_VALUE: {
+            if (service != Message.Servs.s_admin_VALUE) {
+                return ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_fail_service_call_VALUE);
+            }
+
+            byte[] data = parseMsgReq(request, msgHash);
+            Message.req_getBlocksByLatest req;
+
+            try {
+                req = Message.req_getBlocksByLatest.parseFrom(data);
+                Long count = req.getCount();
+                if (count > 1000) {
+                    count = 1000L;
+                }
+
+                Long latest = this.getBestBlock().getNumber();
+                List<Long> numbers = new ArrayList<>();
+                long i = Long.valueOf(latest);
+                while (i > Long.valueOf(latest)-Long.valueOf(count) && i >= 0) {
+                    numbers.add(i);
+                    i--;
+                }
```

Comment:
you can replace with instead of while loop
LongStream.rangeClosed(start, end)
    .boxed().collect(Collectors.toList());

Reply:
please refer
https://stackoverflow.com/questions/10242380/how-can-i-generate-a-list-or-array-of-sequential-integers-in-java/36114723#36114723